### DISCUSSION TITLE
field array data out of sync when submitting after removing one item (via useFieldArray hook)

### DISCUSSION THREAD

**Discussion by devmadou** [react-hook-form/react-hook-form#6527] [Q&A]
First of all, thanks a lot for the great library and all the work you have been doing maintaining it !!

I am trying to implement form edition with data fetched from a database.
I'm leveraging the useFieldArray hook to handle dynamic sections creation and edition.

I created a codesandbox to help you investigate
https://codesandbox.io/s/objective-haze-9vr3x?file=/src/App.js

(I don't know if I am doing something wrong but here are the steps to reproduce my issue)

1. Wait for the data to load (to simulate a fetch request, we simply used a setTimeout and populated the defaultValues with dummy data)
2. Remove the 2nd item of the array
3. Hit the submit button

The data coming into the submit function should normally be an array of one item.

I have been struggling with this issue for quite some time and I would greatly appreciate your help.

**Comment by bluebill1049** [comment]
i have submitted an issue on this: https://github.com/react-hook-form/react-hook-form/issues/6528 will investigate the root cause
